README.md
#  [Glasswidgets](https://pub.dev/packages/glasswidgets)

* A new Flutter package for easy implementation of glassmorphism in flutter apps.

## Getting Started

Currently This package only contains Glasscontainer 

### soon going to add more widgets in this package 
1. glassbutton
2. glassalert
3. glasssnackbar
4. glasscard
etc. 

# screenshots
![s1](https://user-images.githubusercontent.com/37141368/108827077-2563e480-75eb-11eb-9aeb-74e634dd8051.PNG)

## Usage

[Example](https://github.com/yashraut362/Glasswidgets/blob/main/example/lib/main.dart)

To use this package :

* add the dependency to your [pubspec.yaml](https://github.com/KarimMohamed2005/SplashScreenFlutterPackage/blob/master/pubspec.yaml) file.

```yaml
  dependencies:
    flutter:
      sdk: flutter
    glasswidgets:
```

### Simple Examples
# GlassContainer
``` dart
new GlassContainer(
            glasscurve: 30, //giving curve to glassmorphic container
            glassheight: 200, //giving height to glassmorphic container
            glasswidth: 300, //giving width to glassmorphic container
            glasswidget: //Child widget u can use any flutter widget over here.
                Center(
              child: Text(
                'Hello World',
                textAlign: TextAlign.center,
                style: TextStyle(
                    color: Colors.white,
                    fontSize: 20,
                    fontWeight: FontWeight.bold),
              ),
            ),
          ),
```

# GlassSquare

``` dart
new GlassSquare(
                          glassSize: 150,
                          glasscurve: 10,
                          glasswidget: Center(
                            child: Text(
                              'GlassSquare',
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                  color: Colors.white,
                                  fontSize: 20,
                                  fontWeight: FontWeight.bold),
                            ),
                          ),
                        ),
```


# GlassCircle

``` dart
new GlassCircle(
                          glassSize: 150,
                          glasswidget: Center(
                            child: Text(
                              'GlassCircle',
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                color: Colors.white,
                                fontSize: 20,
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                          ),
                        ),
```


CHANGELOG.md
## [0.0.4] - TODO:added GlassSquare and GlassCircle .
* added two new widgets.
* added screenshots

## [0.0.3] - TODO:updated documentation .
* updated readme.

## [0.0.2] - TODO:updated documentation .
* screenshots and description updated.

## [0.0.1] - TODO:Beta -1 .
* added glasscontainer widget.


test/glasswidget_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:glasswidget/glasswidget.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


lib/src/glasscircle.dart
import 'dart:ui';
import 'package:flutter/material.dart';

class GlassCircle extends StatefulWidget {
  GlassCircle({
    Key key,
    this.glassSize,
    this.glasswidget,
  }) : super(key: key);

  final double glassSize;

  final Widget glasswidget;

  @override
  _GlassCircleState createState() => _GlassCircleState();
}

class _GlassCircleState extends State<GlassCircle> {
  @override
  Widget build(BuildContext context) {
    return ClipOval(
      child: Container(
        child: BackdropFilter(
          filter: ImageFilter.blur(sigmaX: 26.0, sigmaY: 26.0),
          child: Container(
              height: widget.glassSize,
              width: widget.glassSize,
              decoration: BoxDecoration(
                color: Colors.white.withOpacity(0.15),
                border: Border.all(
                  width: 1.5,
                  color: Colors.white.withOpacity(0.2),
                ),
              ),
              child: widget.glasswidget),
        ),
      ),
    );
  }
}


lib/src/glasssquare.dart
import 'dart:ui';
import 'package:flutter/material.dart';

class GlassSquare extends StatefulWidget {
  GlassSquare({
    Key key,
    this.glasscurve = 0.0,
    this.glassSize,
    this.glasswidget,
  }) : super(key: key);

  final double glassSize;
  final double glasscurve;
  final Widget glasswidget;

  @override
  _GlassSquareState createState() => _GlassSquareState();
}

class _GlassSquareState extends State<GlassSquare> {
  @override
  Widget build(BuildContext context) {
    return Container(
      child: ClipRRect(
        borderRadius: BorderRadius.circular(widget.glasscurve),
        child: BackdropFilter(
          filter: ImageFilter.blur(sigmaX: 26.0, sigmaY: 26.0),
          child: Container(
              height: widget.glassSize,
              width: widget.glassSize,
              decoration: BoxDecoration(
                color: Colors.white.withOpacity(0.15),
                borderRadius: BorderRadius.circular(widget.glasscurve),
                border: Border.all(
                  width: 1.5,
                  color: Colors.white.withOpacity(0.2),
                ),
              ),
              child: widget.glasswidget),
        ),
      ),
    );
  }
}


lib/src/glasscontainer.dart
import 'dart:ui';
import 'package:flutter/material.dart';

class GlassContainer extends StatefulWidget {
  GlassContainer({
    Key key,
    this.glasscurve = 0.0,
    this.glassheight,
    this.glasswidth,
    this.glasswidget,
  }) : super(key: key);

  final double glassheight;
  final double glasswidth;
  final double glasscurve;
  final Widget glasswidget;

  @override
  _GlassContainerState createState() => _GlassContainerState();
}

class _GlassContainerState extends State<GlassContainer> {
  @override
  Widget build(BuildContext context) {
    return Container(
      decoration: BoxDecoration(boxShadow: [
        BoxShadow(
            blurRadius: 24,
            spreadRadius: 16,
            color: Colors.black.withOpacity(0.2))
      ]),
      child: ClipRRect(
        borderRadius: BorderRadius.circular(widget.glasscurve),
        child: BackdropFilter(
          filter: ImageFilter.blur(sigmaX: 26.0, sigmaY: 26.0),
          child: Container(
              height: widget.glassheight,
              width: widget.glasswidth,
              decoration: BoxDecoration(
                color: Colors.white.withOpacity(0.15),
                borderRadius: BorderRadius.circular(widget.glasscurve),
                border: Border.all(
                  width: 1.5,
                  color: Colors.white.withOpacity(0.2),
                ),
              ),
              child: widget.glasswidget),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:glasswidgets/glasswidgets.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
        debugShowCheckedModeBanner: false,
        home: Scaffold(
            backgroundColor: Colors.black,
            appBar: AppBar(
              backgroundColor: Colors.black87,
              title: Text('GlassWidgets'),
              centerTitle: true,
            ),
            body: Container(
              constraints: BoxConstraints.expand(),
              decoration: BoxDecoration(
                image: DecorationImage(
                    image: AssetImage('assets/bg.jpg'), fit: BoxFit.cover),
              ),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                children: [
                  Center(
                    //Glasscontainer
                    child: GlassContainer(
                      glasscurve: 30, //giving curve to glassmorphic container
                      glassheight:
                          200, //giving height to glassmorphic container
                      glasswidth: 300, //giving width to glassmorphic container
                      glasswidget: //Child widget u can use any flutter widget over here.
                          Center(
                        child: Text(
                          'GlassContainer',
                          textAlign: TextAlign.center,
                          style: TextStyle(
                              color: Colors.white,
                              fontSize: 20,
                              fontWeight: FontWeight.bold),
                        ),
                      ),
                    ),
                  ),
                  Row(
                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                    children: [
                      Center(
                        child: GlassCircle(
                          glassSize: 150,
                          glasswidget: Center(
                            child: Text(
                              'GlassCircle',
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                color: Colors.white,
                                fontSize: 20,
                                fontWeight: FontWeight.bold,
                              ),
                            ),
                          ),
                        ),
                      ),
                      Center(
                        child: GlassSquare(
                          glassSize: 150,
                          glasscurve: 10,
                          glasswidget: Center(
                            child: Text(
                              'GlassSquare',
                              textAlign: TextAlign.center,
                              style: TextStyle(
                                  color: Colors.white,
                                  fontSize: 20,
                                  fontWeight: FontWeight.bold),
                            ),
                          ),
                        ),
                      ),
                    ],
                  )
                ],
              ),
            )));
  }
}
